core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -178,6 +178,46 @@ public void applyTo(TableMetadata.Builder metadataBuilder) {
     }
   }
 
+  class SetStatistics implements MetadataUpdate {
+    private final long snapshotId;
+    private final StatisticsFile statisticsFile;
+
+    public SetStatistics(long snapshotId, StatisticsFile statisticsFile) {
+      this.snapshotId = snapshotId;
+      this.statisticsFile = statisticsFile;
+    }
+
+    public long snapshotId() {
+      return snapshotId;
+    }
+
+    public StatisticsFile statisticsFile() {
+      return statisticsFile;
+    }
+
+    @Override
+    public void applyTo(TableMetadata.Builder metadataBuilder) {
+      metadataBuilder.setStatistics(snapshotId, statisticsFile);
+    }
+  }
+
+  class RemoveStatistics implements MetadataUpdate {
```

Comment:
When is this going to be used? Should we have a single `ReplaceStatistics` operation that is idempotent instead?

Reply:
When working on Trino stats support in Hive and Delta we found out that ability to remove stats is useful. We need a way to remove statistics from a table, for example if we detect that application that wrote stats did it incorrectly, or when stats are correct, but a query engine makes wrong conclusions using those stats and plans are bad in practice. 

@rdblue how would you want to model this?

Reply:
That sounds good to me. Is `SetStatistics` intended to be idempotent? That would make sense so that we don't need both remove and set if we want to replace.

Reply:
Yes, `SetStatistics` should be idempotent and also should allow replacing an existing stat file, so we don't need remove when we want to set.